langchain/agents/conversational_chat/output_parser.py

```diff
@@ -1,31 +1,46 @@
 from __future__ import annotations
 
 import json
+import re
 from typing import Union
 
 from langchain.agents import AgentOutputParser
 from langchain.agents.conversational_chat.prompt import FORMAT_INSTRUCTIONS
 from langchain.schema import AgentAction, AgentFinish
 
+FINAL_ANSWER_PREFIX = '''{
+"action": "Final Answer",
+"action_input": "'''
+
 
 class ConvoOutputParser(AgentOutputParser):
     def get_format_instructions(self) -> str:
         return FORMAT_INSTRUCTIONS
 
     def parse(self, text: str) -> Union[AgentAction, AgentFinish]:
         cleaned_output = text.strip()
-        if "```json" in cleaned_output:
+        cleaned_output = re.sub(r"\n+", "\n", cleaned_output)
+        if "```json" in cleaned_output and "```" in cleaned_output:
             _, cleaned_output = cleaned_output.split("```json")
-        if "```" in cleaned_output:
             cleaned_output, _ = cleaned_output.split("```")
+        elif "```" in cleaned_output:
+            _, cleaned_output, _ = cleaned_output.split("```")
         if cleaned_output.startswith("```json"):
-            cleaned_output = cleaned_output[len("```json") :]
+            cleaned_output = cleaned_output[len("```json"):]
         if cleaned_output.startswith("```"):
-            cleaned_output = cleaned_output[len("```") :]
+            cleaned_output = cleaned_output[len("```"):]
         if cleaned_output.endswith("```"):
             cleaned_output = cleaned_output[: -len("```")]
         cleaned_output = cleaned_output.strip()
-        response = json.loads(cleaned_output)
+        try:
+            response = json.loads(cleaned_output)
+        except:  # Response isn't JSON!
```

Comment:
It's considered bad practice to 'catch the universal `Exception`. JSON decoder errors will typically return a `ValueError`, which includes the `JSONDecodeError` child class.